pkg/store/proxy.go

```diff
@@ -371,52 +475,30 @@ func startStreamSeriesSet(
 				s.warnCh.send(storepb.NewWarnSeriesResponse(errors.New(w)))
 				continue
 			}
-
+			metrics.recvChannelSize.WithLabelValues(storeAddr).Observe(float64(len(s.recvCh)))
```

Comment:
it's better to use prepared metrics with preset labels here too. Also - it will be much faster